modules/network/src/main/java/org/apache/ignite/internal/network/netty/ConnectionManager.java

```diff
@@ -105,8 +107,7 @@ public class ConnectionManager implements ChannelCreationListener {
     /** Node consistent id. */
     private final String consistentId;
 
-    /** Node launch id. As opposed to {@link #consistentId}, this identifier changes between restarts. */
-    private final UUID launchId;
+    private volatile @Nullable ClusterNode localNode;
```

Comment:
Why is this field volatile now?

Reply:
It is not final. I'm not sure we have a happens-before between the startup of `ConnectionManager` and establishment of connections. Also, this field is used rarely (each time when a connection is established), so the performance penalty should be negligible.

Reply:
My question was about why is this field no longer final)

Reply:
Before building that ClusterNode, we need to start the ConnectionManager to obtain a bound server socket from which we take local address, and then apply some transformations to it. The starting is made in the `start()` method. So the field cannot be final.